### DISCUSSION TITLE
Configuring allowed_policy_member_domains exceptions for specific projects

### DISCUSSION THREAD

**Discussion by markusl** [GoogleCloudPlatform/cloud-foundation-fabric#2613] [General]
We have some projects in our GCP organization that require allowing allowed_policy_member_domains while we would still need to keep the setting for the rest of the projects. The configuration is in `terraform.tfvars` in `org_policies_config` section.

Can you please advise what the recommended way to proceed would be?

Environment: Terraform v1.9.7

## To Reproduce ##

Install the project.

## Expected behavior## 

The project provides an ability to make an exception to the rule on per project basis.

## Result ##

![image](https://github.com/user-attachments/assets/43807cab-3d80-4ae9-a06f-447dbdd561af)

Adding this to a project file

```
tag_bindings:
  allowed-policy-member-domains-all: "tagValues/23456783435345"
```
also results in an error:
```
│ Error: Error creating TagBinding: googleapi: Error 403: The caller does not have permission

```

**Comment by ludoo** [comment]
The service account doing the tag binding needs the tag viewer role on both the tag value and the node where the tag is applied. Is this in the FAST project factory?

**Reply by markusl** [reply]
Yes, this is created using FAST project factory. Allowing external access with similar config does work just fine and we use it for a couple of projects:

```
tag_bindings:
  compute-allow-external-access: "tagValues/281479237543882"
```

I'm trying to understand how these are different from each other?

**Reply by ludoo** [reply]
struggling to understand what "these" means :) anyway, assign the role mentioned above and try again

**Reply by markusl** [reply]
The service account we use is the one created by the FAST framework: `dev-resman-pf-0@iac-core-0.iam.gserviceaccount.com`

The project factory is documented to support `tag_bindings`: https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/tree/master/modules/project-factory

Creating this tag from `org-policies` works just fine:
```
tag_bindings:
  compute-allow-external-access: "tagValues/281479237543882"
```

But creating `allowed-policy-member-domains-all` does not.

I see this is because we have configured `roles/resourcemanager.tagUser` for our service accounts manually in `terraform.tfvars` in `0-bootstrap` stage. However, `allowed_policy_member_domains` is handled differently somehow internally and has different settings and we would like to understand what is the recommended way to configure it so that an exception can be made per-project when needed.

The current config looks like:

```
org_policies_config = {
  constraints = {
    allowed_policy_member_domains = ["org-id"]
  }
}
```
There does not seem to be a way to configure it in similar way other `tag_values` are configured.

**Reply by ludoo** [reply]
tag values behave identically, make sure the service account has tag user role on both tag value and destination (folder or project)

**Reply by markusl** [reply]
Obviously they do behave identically if they are configured identically. Let me be clear about the question we are having :D 

In FAST framework configuration code, `allowed_policy_member_domains` is configured differently from other tags which are specified in `tag_values`.

When using `tag_values`, we are able to add the `resourcemanager.tagUser` in the `iam` section.
`allowed_policy_member_domains` is configured in `constraints` block which has a type of 

```
    constraints = optional(object({
      allowed_policy_member_domains = optional(list(string), [])
    }), {})
```

The question: How do we configure the IAM section for `allowed_policy_member_domains` when using the `constraints` block?
Should we migrate the tag value away from `allowed_policy_member_domains` completely? How to do that?

**Reply by ludoo** [reply]
ok I get your point :) it's different because it's an example we provide out of the box, but you can override its IAM via the `tags` variable

https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/fast/stages/0-bootstrap/organization.tf#L293

**Reply by markusl** [reply]
I didn't expect it to work, but it indeed does work and merges the values appropriately. 

Thank you for the quick responses and have a nice day!

**Reply by ludoo** [reply]
thanks for confirming, and sorry for taking some time to get there :)
